expression/builtin_op.go

```diff
@@ -352,25 +352,26 @@ type unaryMinusFunctionClass struct {
 }
 
 func (b *unaryMinusFunctionClass) handleIntOverflow(arg *Constant) (overflow bool) {
-	overflow = false
 	if mysql.HasUnsignedFlag(arg.GetType().Flag) {
 		uval := arg.Value.GetUint64()
 		// -math.MinInt64 is 9223372036854775808, so if uval is more than 9223372036854775808, like
 		// 9223372036854775809, -9223372036854775809 is less than math.MinInt64, overflow occurs.
 		if uval > uint64(-math.MinInt64) {
-			overflow = true
+			return true
 		}
 	} else {
 		val := arg.Value.GetInt64()
 		// The math.MinInt64 is -9223372036854775808, the math.MaxInt64 is 9223372036854775807,
 		// which is less than abs(-9223372036854775808). When val == math.MinInt64, overflow occurs.
 		if val == math.MinInt64 {
-			overflow = true
+			return true
 		}
 	}
-	return
+	return false
 }
 
+// typeInfer infer unary Minus Function return type. when arg is int constant and overflow,
```

Comment:
s/ infer/ infers
s/ unary Minus/ unaryMinus
s/ Function/ function
s/ when arg/ When the arg
s/ int/ an int